### ISSUE TITLE
Missing callables in scene for diff-rt-calibration forward pass

### ISSUE THREAD

**Issue by naperman** [NVlabs/sionna#794]
This issue is a follow-up of the question asked [here](https://github.com/NVlabs/sionna/discussions/776#discussioncomment-12560725) and it's response by @faycalaa 

Specifically, I want to understand which properties of the new Sionna 1.0 [_scene_](https://github.com/NVlabs/sionna-rt/blob/main/src/sionna/rt/scene.py) object should be initialized using [TrainableMaterials](https://github.com/NVlabs/diff-rt-calibration/blob/main/code/trainable_materials.py) and [TrainableScatteringPattern](https://github.com/NVlabs/diff-rt-calibration/blob/main/code/trainable_scattering_pattern.py) to replace the following lines in the train function in [Synthetic_Data.ipynb](https://github.com/NVlabs/diff-rt-calibration/blob/main/notebooks/Synthetic_Data.ipynb) example.

```
def train(train_dict):
    optimizer = tf.keras.optimizers.Adam(5e-3)

    # Apply a trainable antenna pattern
    scene.tx_array.antenna.patterns[0] = TrainableAntennaPattern(num_mixtures=3)

    # Use a trainable scattering pattern
    scene.scattering_pattern_callable = TrainableScatteringPattern()

    # Replace the current radio materials by learned ones
    scene.radio_material_callable = TrainableMaterials(scene, num_objects=len(scene.objects), embedding_size=30)
```

The new Sionna 1.0 [_scene_](https://github.com/NVlabs/sionna-rt/blob/main/src/sionna/rt/scene.py) object is missing [radio_material_callable](https://jhoydis.github.io/sionna-0.19.2-doc/api/rt.html#sionna.rt.Scene.radio_material_callable) and [scattering_pattern_callable](https://jhoydis.github.io/sionna-0.19.2-doc/api/rt.html#sionna.rt.Scene.scattering_pattern_callable) that were present in it's v0.19.2 and older equivalents.

**Comment by faycalaa** [comment]
Hi @naperman,

Training of material properties can be done with Sionna 1. However, you would need to adapt the code to the new API of Sionna 1, as the [diff-rt](https://github.com/NVlabs/diff-rt) repository is currently only compatible with previous versions of Sionna. This means that the `TrainableMaterials` and `TrainableScatteringPattern` classes cannot be reused with Sionna 1.

You should check out the developer guides on [radio materials](https://nvlabs.github.io/sionna/rt/developer/dev_custom_radio_materials.html#understanding-radio-materials) to understand how to get gradients with respect to material properties with Sionna 1. Note that other elements presented in the [paper corresponding to the diff-rt repository](https://ieeexplore.ieee.org/abstract/document/10705152), such as over-parametrization of trainable material properties, are not illustrated in the developer guides but could be implemented using DrJit.
